utils.mk

```diff
@@ -3,6 +3,27 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+# Owner for installed files
+INSTALL_OWNER     ?= root
+
+# Group for installed files
+INSTALL_GROUP     ?= adm
+
+# Permissions for installed executable files
+INSTALL_EXE_PERMS ?= 0750
```

Comment:
CI fails with:

```
20:43:23 .ci/install_kata.sh: line 56: /usr/local/bin/kata-runtime: Permission denied
```

The mod 750 means others can't execute these binaries.

Reply:
and this may clash with the rootless use case .. :thinking:  :question: 

cc @fengwang666

Reply:
That's true. But I think that's what we should be encouraging. The reason is that the configuration file could contain sensitive information such as [Jaeger credentials](https://github.com/kata-containers/kata-containers/commit/340650270658d2e6ae0a3e52c1865c0bb4b81acf). That implies that "other" should not be able to read the file. But when you run `kata-runtime`, the code will read the configuration file. Overall, that means that admins should not allow non-priv / "other" users to run `kata-runtime`. We could keep the group ownership as `root`, but by using `adm` admins can give that group membership to some users without giving them full `root` privs. In summary, the behaviour of this PR is that you would need to be in the `INSTALL_GROUP` group to be able to run any Kata commands.

This changes makes the install "safe by default". Again if packagers / distro wish to change that behaviour, they can run `make INSTALL_EXE_PERMS="..."` as they wish.



That's right, and I think that's actually how users should be running it.

Reply:
This means we would at least add some supplemental steps for validation, e.g. creating a `kata` group? Is that the intent?

Reply:
Hi @c3d - I'm not sure I follow? The intent of this PR is to improve the installation of files and make them as secure as possible by default. A `kata` group _could_ be used but I'm suggesting we use the `adm` group as the default as that should already exist. Having a `kata` group would be better imho in the longer term as that would give admins more scope and flexibility. And for some context lthough containerd doesn't provide it's own group, docker does.

An alternative to `adm` group ownership and clearing all the "other" perms and allowing non-priv users to run `kata-runtime` as they can currently could be to rework the config handling so that we store sensitive config details in a separate (more secure) file, but that's a much bigger change than this, and I'm personally not convinced about that idea currently.

Reply:
Sorry @jodh-intel I had not refreshed my browser and your comment did not show up. I was replying directly to @devimc

Reply:
@c3d that's a good question, we may need to add `kata-$RANDOM` user to the `adm` group, but the most important thing here is that the job that was testing rootless has been removed/disabled, **the CI is not testing rootless** :exclamation::exclamation:
 
:speaker: @fengwang666

Reply:
@devimc As long as the qemu executable permission is not changed, the the rootless hypervisor feature shouldn't be impacted. The [rootless](https://github.com/kata-containers/tests/blob/main/.ci/run.sh#L68) CI seems still running?

Reply:
@fengwang666 

> The [rootless](https://github.com/kata-containers/tests/blob/main/.ci/run.sh#L68) CI seems still running?

no, there is no job exporting `CRIO_K8S` variable, that chunk of code is useless https://github.com/kata-containers/tests/blob/main/.ci/run.sh#L64-L70

Reply:
Hi @devimc ,

It should be running on the baseline jobs (run daily). Example:
http://jenkins.katacontainers.io/view/Daily%20baseline/job/kata-containers-main-fedora-35-x86_64-CRIO_K8S-baseline/

And in jobs triggered by pull requests. Example:
http://jenkins.katacontainers.io/view/PR%20Triggered/job/kata-containers-fedora-35-x86_64-CRIO_K8S-PR/

Reply:
Just to make it clear, there are two scenarios for rootless - rootless hypervisor and rootless runtime itself when the upper layers such as containerd/podman itself call kata as a non-priveleged user.
I am afraid that this PR will break the later case, as we restrict kata-runtime to be called by restricted user group. 
We do not have control over what user calls kata-runtime in that case.

My suggestion would be to have kata-runtime as world-executable and the kata conf file as world-readable. I believe any sensitive information should not be part of config files itself.  I know this exposes any sensitive information present in the kata-configuration file, but I don't think they should have been there in any place. We could have another file to hold any sensitive data and that could be restricted to be read on by admin/root group.

cc @jodh-intel

Reply:
@amshinde - I'm a little confused. This PR is trying to resolve a permissions issue to ensure that if the admin enters sensitive information in the configuration file the effect is limited since only privileged users would be able to access that configuration file. Are you proposing that we just close this PR and come up with an alternative mechanism for storing the sensitive details? If so, _maybe_ https://github.com/kata-containers/kata-containers/pull/4109 could be used/abused for this purpose since you could then store just the sensitive info in a `secrets.toml` or similar. However, if a non-priv user runs `kata-runtime` currently, the (entire) configuration needs to be available to that user. This PR resolves that issue by requiring the user running the command to be privileged, but if I'm understanding what you're saying, we'd need to make the fragments PR "tolerate" unreadable files by just ignoring them, to handle the scenario where a system has sensitive information stored in a config fragment.

Reply:
> The question is whether the resulting semantics aren't a bit too subtle and whether it wouldn't invite unexpected problems 

Agreed. I really don't think that random thought is a good idea, it was more to try to understand @amshinde's comments.

> handling ordinary and sensitive data seems just fundamentally different. 

Again, I agree. I think it was a mistake to allow sensitive data to be stored in our config file. However, we do allow it so this PR was the most immediate solution to the problem using simple primitives. But that doesn't mean we cannot resolve this in other more secure ways in future given additional engineering effort.

@amshinde - if you wish to look into an alternative scheme, that's perfectly fine by me. However, it is worth pointing out that _by default_ neither this PR nor a more elaborate scheme to store secrets is necessary since by default no secrets are stored in the configuration file. It's also worth stressing that if we allow non-priv users to continue to run the `kata-runtime` binary, we're going to hit complications if we store the secrets in a different location or use a different method since (atleast currently) the entire configuration needs to be read by the runtime at startup. I guess we could consider having `kata-runtime` and `containerd-shim-kata-v2` special-case reading secrets based on the uid of the caller ("If running non-priv don't load secrets, else do"). However special casing like this isn't ideal imho and would need careful testing.